api/lib/opentelemetry/instrumentation/base.rb

```diff
@@ -180,12 +186,31 @@ def install(config = {})
         return true if installed?
         return false unless installable?(config)
 
-        @config = config unless config.nil?
+        @config = config_options(config)
         instance_exec(@config, &@install_blk)
         @tracer ||= OpenTelemetry.tracer_provider.tracer(name, version)
         @installed = true
       end
 
+      def config_options(config_hash)
+        config_hash ||= {}
+
+        @options.each_with_object({}) do |option, h|
+          option_name = option[:name]
+          config_hash_value = config_hash[option_name]
+
+          value = if config_hash_value.nil? && option[:allow_nil]
+            nil
+          elsif option[:validate].call(config_hash_value)
+            config_hash_value
```

Comment:
So is the idea that the validation proc itself would log a warning? or, should we attempt to log something uniform here?

Reply:
>imo validation proc should just return true/false
I agree with what you mentioned above that the proc should only return true/false.  I think that we should only be logging an error/warning/info (to be discussed) in the event that the validation fails on user provided config options.  

When no option is provided and the default is used I think that's a quiet happy path.

Reply:
> should we attempt to log something uniform here?

I believe we should be responsible for the logging and not the proc.